src/nvim/ui_client.c

```diff
@@ -81,20 +86,30 @@ uint64_t ui_client_start_server(int argc, char **argv)
   return channel->id;
 }
 
+static void on_stdio_close(Stream *stream, void *data)
+  FUNC_ATTR_NORETURN
+{
+  // Channel *chan = data;
+  ELOG("xxxxxxxxxxxxxxxxxxxxxxxxxxx");
+  os_exit(0);
```

Comment:
The parent (`--remote-ui` client) exits itself when the stdio channel is closed.